### ISSUE TITLE
"Could not build module Foundation" error when compiling for iOS with SDK 15.2

### ISSUE THREAD

**Issue by alexzheng** [godotengine/godot#56092]
### Godot version

3.4.1

### System information

Mac 12.0.1 

### Issue description

Compiling for iOS using 

scons p=iphone tools=no target=release arch=arm64

get lots of this kinds of errors.

In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.2.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/UserNotifications.h:8:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.2.sdk/System/Library/Frameworks/UserNotifications.framework/Headers/NSString+UserNotifications.h:8:9: fatal error: could not build module 'Foundation'
#import <Foundation/Foundation.h>

In file included from /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.2.sdk/System/Library/Frameworks/CoreGraphics.framework/Headers/CGBase.h:8:
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.2.sdk/usr/include/c++/v1/stdbool.h:28:15: fatal error: could not build module 'Darwin'
#include_next <stdbool.h>

### Steps to reproduce

Compiling for iOS using 

scons p=iphone tools=no target=release arch=arm64

**Comment by akien-mga** [comment]
CC @godotengine/ios

**Comment by alexzheng** [comment]
not only Foundation

Similar errors for other modules:

/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.2.sdk/System/Library/Frameworks/CoreServices.framework/Headers/UTCoreTypes.h:19:10: fatal error: could not build module 'CoreFoundation'

**Comment by bruvzg** [comment]
I can't reproduce it, building `3.4` and `3.x` branches works fine with SDK 15,2 (19C51).

`Foundation` and `CoreFoundation` are part of the Objective-C standard library, probably your Xcode installation is broken, this sometimes happens when updating Xcode from App Store. Check if it's not updating in the background (Xcode updates can take hours). If it’s not, try reinstalling Xcode (delete it, reboot and reinstall).

**Comment by alexzheng** [comment]
What's version of scons you used.
I can build 3.x before update the scons.

**Comment by bruvzg** [comment]
> What's version of scons you used.

```
SCons by Steven Knight et al.:
	SCons: v4.3.0.559790274f66fa55251f5754de34820a29c7327a, Tue, 16 Nov 2021 19:09:21 +0000, by bdeegan on octodog
	SCons path: ['/opt/homebrew/Cellar/scons/4.3.0/libexec/lib/python3.10/site-packages/SCons']
```

**Comment by alexzheng** [comment]
My version is similar.
I can build any exported godot project in Xcode, and the it has update to Version 13.2.1 (13C100)

**Comment by alexzheng** [comment]
And I can build for Mac 
scons platform=osx arch=x86_64 --jobs=$(sysctl -n hw.logicalcpu)

**Comment by alexzheng** [comment]
Sorry, the error happens when build for x86_64, it can build for arm64.

scons p=iphone tools=no target=release arch=x86_64 --jobs=$(sysctl -n hw.logicalcpu)

reinstall Xcode, still the same. here is the errors

[errors.txt.zip](https://github.com/godotengine/godot/files/7746344/errors.txt.zip)

**Comment by bruvzg** [comment]
If I remember correctly, building for `x86_64` require `ios_simulator=yes` option.

**Comment by alexzheng** [comment]
This option works, thanks.

**Comment by akien-mga** [comment]
Maybe we can add an error when `arch=x86_64` is passed without `ios_simulator=yes`.

**Comment by alexzheng** [comment]
It can build without this option in the past, may be release 3.2, I can not remember.
And the document  also need update.

**Comment by bruvzg** [comment]
This option was added to allow building both `arm64` device (real iOS) and simulator (running in the simulator on M1 macs). Previously it was always building for device if arch is `arm64` and for the simulator if it's `x86_64`.

We should enable it by default if `arch` is `x86_64`.

**Comment by TedSpikes** [comment]
Hello everyone,

I wanted to pick this one up as my first issue. The fix seems simple enough, but I'm not sure if I put it in exactly the right location/worded the error message in the right way. Let me know what you think!

PR: #65648.

P.S. I think crashing here is better than modifying arguments after the command is already run. That's me though, I'd love to know what the conventions are. Thank you.

**Comment by akien-mga** [comment]
Fixed by #65648.
